### PR TITLE
CVE patch

### PR DESCRIPTION
**Pull request by nnmin-aws**
golang.org/x/net(1.version: v0.7.0 2.package manager: GOBINARY 3.file path: aws-encryption-provider) | MEDIUM | https://nvd.nist.gov/vuln/detail/CVE-2023-3978

golang.org/x/net(1.version: v0.7.0 2.package manager: GOBINARY 3.file path: aws-encryption-provider) | NA | https://nvd.nist.gov/vuln/detail/CVE-2023-39325

google.golang.org/grpc(1.version: v1.49.0 2.package manager: GOBINARY 3.file path: aws-encryption-provider) | NA | https://nvd.nist.gov/vuln/detail/CVE-2023-44487

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -8,8 +8,8 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ARG BUILDER=public.ecr.aws/eks-distro-build-tooling/golang:1.20.10-11-gcc
-ARG BASE_IMAGE=public.ecr.aws/eks-distro/kubernetes/go-runner:v0.13.0-eks-1-23-9
+ARG BUILDER=public.ecr.aws/eks-distro-build-tooling/golang:1.21.5-gcc
```
Comment (xdu31) [inline Dockerfile:11]: You can just bump the `.go-version` in https://github.com/kubernetes-sigs/aws-encryption-provider/blob/master/.go-version

**Review by micahhausler — APPROVED** [review]
/lgtm
/approve

**Pull request merged**